Kernel/Devices/VMWareSVGADevice.h

```diff
@@ -0,0 +1,262 @@
+    virtual KResultOr<size_t> read(FileDescription&, u64, UserOrKernelBuffer&, size_t) override { return -EINVAL; }
+    virtual KResultOr<size_t> write(FileDescription&, u64, const UserOrKernelBuffer&, size_t) override { return -EINVAL; }
+
+    virtual void handle_irq(const RegisterState&) override;
+
+    enum class IORegister {
+        ID = 0,
+        Enable,
+        Width,
+        Height,
+        MaxWidth,
+        MaxHeight,
+        Depth,
+        BitsPerPixel,
+        PseudoColor,
+        RedMask,
+        GreenMask,
+        BlueMask,
+        BytesPerLine,
+        FBStart,
+        FBOffset,
+        VRamSize,
+        FBSize,
+        Capabilities,
+        MemStart,
+        MemSize,
+        ConfigDone,
+        Sync,
+        Busy,
+        GuestID,
+        CursorID,
+        CursorX,
+        CursorY,
+        CursorOn,
+        HostBitsPerPixel,
+        ScratchSize,
+        MemRegs,
+        NumDisplays,
+        Pitchlock,
+        IRQMask,
+        NumGuestDisplays,
+        DisplayID,
+        DisplayIsPrimary,
+        DisplayPositionX,
+        DisplayPositionY,
+        DisplayWidth,
+        DisplayHeight,
+        GMRID,
+        GMRDescriptor,
+        GMRMaxIDS,
+        GMRMaxDescriptorLength,
+        Traces
+    };
+
+    enum class FifoRegister {
+        Min = 0,
+        Max,
+        NextCMD,
+        Stop,
+        Capabilites = 4,
+        Flags,
+        Fence,
+        SVGA_FIFO_3D_HWVERSION,
+        Pitchlock,
+        CursorOn,
+        CursorX,
+        CursorY,
+        CursorCount,
+        CursorLastUpdated,
+        ReservedBytes,
+        CursorScreenID,
+        _3DCaps = 32,
+        _3DCapsLast = 32 + 255,
+        Guest3DHWVersion,
+        FenceGoal,
+        Busy,
+        NumRegs
+    };
+
+    enum class Capabilities {
+        RectCopy = 0x2,
+        Cursor = 0x20,
+        CursorBypass = 0x40,
+        CursorBypass2 = 0x80,
+        _8BitEmulation = 0x100,
+        AlphaCursor = 0x200,
+        _3D = 0x4000,
+        ExtendedFifo = 0x8000,
+        MultiMon = 0x10000,
+        Pitchlock = 0x20000,
+        IRQMask = 0x40000,
+        DisplayTopology = 0x80000,
+        GMR = 0x100000,
+        Traces = 0x200000
+    };
+
+    enum class FifoCapabilites {
+        Fence = 1,
+        AccelFront = 1 << 1,
+        Pitchlock = 1 << 2,
+        Video = 1 << 3,
+        CursorBypass3 = 1 << 4,
+        Escape = 1 << 5,
+        Reserve = 1 << 6,
+        ScreenObject = 1 << 7
+    };
+
+    enum class Commands {
+        Invalid = 0,
+        Update,
+        RectCopy = 3,
+        DefineCursor = 19,
+        DefineAlphaCursor = 22,
+        UpdateVerbose = 25,
+        FrontRopFill = 29,
+        Fence = 30,
+        Escape = 33,
+        DefineScreen = 34,
+        DestroyScreen,
+        DefineGMRFB,
+        BlitGMRFBToScreen,
+        BlitScreenToGMRFB,
+        AnnotationFill,
+        AnnotationCopy,
+        Max
+    };
+
+    enum class Ports {
```

Comment:
```suggestion
    enum class Ports: u32 {
```
You cast this enum (and a few other enums) to u32 using `static_cast<u32>`. As such, it's probably a good idea to explicitly declare them as u32 in the first place !